.github/workflows/codeql.yml

```diff
@@ -0,0 +1,53 @@
+name: "CodeQL Analysis"
+permissions: {}
+
+on:
+  push:
+    branches: [ "main"]
+  pull_request:
+    branches: [ "main" ]
+  schedule:
+    # Run at 00:36 AM every Wednesday
+    - cron: '36 0 * * 3'
+
+jobs:
+  analyze:
+    name: Analyze Python code with CodeQL
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+    permissions:
+      # Required for all workflows
+      security-events: write
+      # Required to fetch internal or private CodeQL packs
+      packages: read
+      # Required for workflows in private repositories
+      actions: read
+      contents: read
+
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v5
+
+    - name: Set up Python
+      uses: actions/setup-python@v6
+      with:
+        python-version: '3.9'
+
+    - name: Install dependencies
```

Comment:
install deps the proper way by intalling the package instead?

(See the other actions)